packages/account-v2/src/modules/Onfido/OnfidoContainer.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from 'react';
+import clsx from 'clsx';
+import { useHistory } from 'react-router-dom';
+import { useOnfido } from '@deriv/api';
+import { qtMerge } from '@deriv/quill-design';
+import { Button, Loader, Text } from '@deriv-com/ui';
+import IcAccountMissingDetails from '../../assets/proof-of-identity/ic-account-missing-details.svg';
+import { ErrorMessage } from '../../components/ErrorMessage';
+import { IconWithMessage } from '../../components/IconWithMessage';
+import { OnfidoView } from '../../containers/Onfido/OnfidoView';
+
+// TODO: Remove optional and default props when POI is ready
+type TOnfidoContainer = {
+    country?: string;
+    isEnabledByDefault?: boolean;
+    onOnfidoSubmit?: () => void;
+};
+
+export const OnfidoContainer = ({ country = 'co', isEnabledByDefault = false, onOnfidoSubmit }: TOnfidoContainer) => {
+    const [isOnfidoEnabled, setIsOnfidoEnabled] = useState(isEnabledByDefault);
+    const [transitionEnd, setTransitionEnd] = useState(false);
+    const history = useHistory();
+
+    const {
+        data: { hasSubmitted, onfidoContainerId, onfidoRef },
+        isOnfidoInitialized,
+        isServiceTokenLoading,
+        onfidoInitializationError,
+        serviceTokenError,
+    } = useOnfido(country);
+
+    useEffect(() => {
+        if (hasSubmitted) {
+            onOnfidoSubmit?.();
+            onfidoRef?.current?.safeTearDown();
+        }
+    }, [hasSubmitted, onfidoRef, onOnfidoSubmit]);
+
+    useEffect(() => {
+        let transitionEndTimeout: ReturnType<typeof setTimeout>;
+        if (isOnfidoEnabled) {
+            transitionEndTimeout = setTimeout(() => setTransitionEnd(true), 350);
+        }
+        return () => {
+            if (transitionEndTimeout) {
+                clearTimeout(transitionEndTimeout);
+            }
+        };
+    }, [isOnfidoEnabled]);
+
+    const showMissingPersonalDetails = ['MissingPersonalDetails', 'InvalidPostalCode'].includes(
```

Comment:
can we rename this variable to "hasPersonalDetailsValidationErrors" or something as we are checking both MissingPersonalDetails and InvalidPostalCode in this condition

Reply:
Updated.